### PR TITLE
onMath: filter out "i'm g"

### PR DESCRIPTION
**Pull request by euank**
Prior to this, "i'm g" -> "(i) m g".

After this change, "i'm g" no longer expands.

I also added a bunch of tests because frankly this is ridiculous.

r? @LinuxMercedes @hfukada @drusepth

### CODE REVIEW

**Comment by hfukada** [comment]
:fisi:

File: modules/onMath/onmath.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+describe('onMath', function() {
+  it('should do simple math', function(done) {
+    var simpleMath = [
+      ["1+1", "2"],
+      ["1*1", "1"],
+    ];
+
+    async.every([["1 + 1", "2"], ["1 * 1", "1"]], function(el, callback) {
```
Comment (LinuxMercedes) [inline modules/onMath/onmath.test.js]: should this line be using `simpleMath`?

File: modules/onMath/onmath.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  });
+  it("should ignore dumb stuff", function(done) {
+    var dumbStuff = [
+      "1",
+      "e",
+      "E",
+      "i'm g",
+      "9/10",
+      "10/10",
+      "10/100",
```
Comment (LinuxMercedes) [inline modules/onMath/onmath.test.js]: To be fully meme-compatible we should include 11/10 here

File: modules/onMath/onmath.test.js (hunk 1 above)
Comment (euank) [inline modules/onMath/onmath.test.js]: good catch, thanks

**Comment by euank** [comment]
@LinuxMercedes addressed comments; the overall change LGTY, right?

**Comment by LinuxMercedes** [comment]
:star: :star: :star: :star: :star:

:+1: :racehorse: :penguin:

**Pull request merged**